en/docs/reference/mediators/cache-mediator.md

```diff
@@ -156,90 +156,102 @@ Following are examples of how you can use the Cache mediator.
-When a similar message is sent to the endpoint for the second time, the
-previous response is directly fetched from the cache and sent to the
-requester. This happens because the `         onCacheHit        `
-sequence is not defined in this configuration.
+In this example configuration, when the first message is sent to the endpoint, 
+the cache is not hit because it is the initial request and the cache is empty. 
+The Cache mediator, configured in the main sequence, caches the response to this message. 
+When a similar message is sent to the endpoint for the second time, the response is directly fetched from the cache 
+and sent to the requester. This caching behavior by the onCacheHit element defined within the Cache mediator 
+configuration. The onCacheHit element specifies that a cached response should be immediately returned if a cache hit 
+occurs, without forwarding the request to the backend service again. This mechanism significantly improves 
+the performance for repeated requests with the same cache key.
 
-``` java
-<?xml version="1.0" encoding="UTF-8"?>
-<sequence name="main">
-        <in>
-            <cache collector="false" maxMessageSize="10000" timeout="20">
-                <protocol type="HTTP">
-                    <methods>POST</methods>
-                    <headersToExcludeInHash/>
-                    <responseCodes>2[0-9][0-9]</responseCodes>
-                    <enableCacheControl>false</enableCacheControl>
-                    <includeAgeHeader>false</includeAgeHeader>
-                    <hashGenerator>org.wso2.carbon.mediator.cache.digest.HttpRequestHashGenerator</hashGenerator>
-                </protocol>
-                <implementation maxSize="100"/>
-            </cache>
-            <send>
-                <endpoint name="inlined">
-                    <address uri="http://localhost:9000/services/SimpleStockQuoteService"/>
-                </endpoint>
-            </send>
-        </in>
-        <out>
-            <cache collector="true"/>
-            <send/>
-        </out>
+=== "Sequence"
+    ``` xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <sequence name="main" trace="disable" xmlns="http://ws.apache.org/ns/synapse">
+        <cache id="simpleCache" scope="per-host" collector="false" hashGenerator="org.wso2.carbon.mediator.cache.digest.DOMHASHGenerator" timeout="20" maxMessageSize="10000">
+            <implementation type="memory" maxSize="100"/>
+            <onCacheHit>
+                <respond/>
+            </onCacheHit>
+            <protocol type="HTTP">
+                <methods>POST</methods>
+                <headersToExcludeInHash/>
+                <responseCodes>2[0-9][0-9]</responseCodes>
+                <enableCacheControl>false</enableCacheControl>
+                <includeAgeHeader>false</includeAgeHeader>
+            </protocol>
+        </cache>
+        <call>
+            <endpoint key="SimpleStockQuoteEndpoint"/>
+        </call>
```

Comment:
A respond mediator should be needed here.

Reply:
Shall I wrap this sequence inside an API? I left the sequence's name as `main`, Shall I change it to avoid confusions with previously existed main sequence?